### PR TITLE
Add temporal response in django's get context

### PR DESCRIPTION
**Pull request by patrick91**
Similar to #750 but for django 😊

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release updates get_context in the django integration to also receive a temporal response object that can be used to set headers, cookies and status code.


```
@strawberry.type
class Query:
    @strawberry.field
    def abc(self, info: Info) -> str:
        info.context.response.status_code = 418

        return "ABC"
```

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Release type: minor
```
Comment (alex-verve) [inline RELEASE.md:1]: Won't it be braking change for people who already inherit from  `GraphQLView` and expecting only one required argument in  `.get_context(request)` method?

File: strawberry/django/views.py [hunk 2]
```diff
@@ -87,13 +88,31 @@ def _render_graphiql(self, request: HttpRequest, context=None):
...
+        self, response_data: GraphQLHTTPResponse, sub_response: HttpResponse
+    ) -> JsonResponse:
+        response = JsonResponse(response_data)
+        response._headers.update(sub_response._headers)
+
+        # unfortunately setting None as status code doesn't work
+        # so to make sure we only override status codes when they have been
+        # changed we use a status code that doesn't actually exist
+        # https://github.com/django/django/blob/03d0f12/django/http/response.py#L117-L124
+        if sub_response.status_code != 199:
```
Comment (alex-verve) [inline strawberry/django/views.py:101]: Do you think if it's worth using some private constant for this integer?

**Review by alex-verve — APPROVED** [review]
🥇

File: RELEASE.md (hunk 1 above)
Comment (patrick91) [reply RELEASE.md:1]: yes, it is breaking 😊 we are still pre 1.0 so it's fine to just bump to minor :)

File: strawberry/django/views.py (hunk 2 above)
Comment (patrick91) [reply strawberry/django/views.py:101]: that's a good suggestion, we could also:

1. have a subclass of httpresponse that allows to set status_code to none (or uses None as default)
2. set status_code as none after creating the instance

let me try 1

Comment (patrick91) [reply strawberry/django/views.py]: @alex-verve 1 worked well 😊

**Comment by AGASS007** [comment]
there's a new API for headers in 3.2

https://github.com/django/django/pull/13186

I guess we'll have to either differentiate based on django version or catch an `AttributeError`.

**Comment by patrick91** [comment]
@AGASS007 thanks for pointing that out, I guess the current way will work for a while, right?

**Comment by AGASS007** [comment]
hum, django 3.2 will be released in 2 weeks (rc1 is out today)

File: strawberry/django/views.py [hunk 3]
```diff
@@ -23,6 +24,10 @@
 from .context import StrawberryDjangoContext
 
 
+class TemporalHttpResponse(HttpResponse):
```
Comment (alex-verve) [inline strawberry/django/views.py:27]: I just realized it's better to inherit from `JsonResponse` here.
As `Content-Type` header will be overwritten  from json to `text/html` in line 99 below

Comment (patrick91) [reply strawberry/django/views.py:27]: good spot! sending a PR now

**Comment by blueyed** [comment]
Note that `status_code = None` makes Django's `HttpResponse.__repr__` crash: https://github.com/django/django/pull/15768 (just for info/reference, but maybe `__repr__` needs to (or even should?) be overridden/adopted here).

**Pull request merged**